File: code/backend/python/app/core/ports/story/story_import_port.py
from abc import ABC, abstractmethod
from typing import Dict, Any, List
from app.core.models.story.story_import_result import StoryImportResult

class StoryImportPort(ABC):
    @abstractmethod
    def import_story(self, data: Dict[str, Any]) -> StoryImportResult:
        pass

    @abstractmethod
    def delete_story(self, uuid: str) -> bool:
        pass

File: code/backend/php/src/Core/Port/Story/StoryImportPort.php
<?php

declare(strict_types=1);

namespace Games\Paths\Core\Port\Story;

use Games\Paths\Core\Domain\Story\StoryImportResult;

interface StoryImportPort
{
    public function importStory(array $data): StoryImportResult;

    public function deleteStory(string $uuid): bool;
}
